node/core/approval-voting/src/lib.rs

```diff
@@ -1248,6 +1250,8 @@ async fn handle_approved_ancestor(
 			Some(b) => b,
 		};
 
+		candidates.push((block_hash, entry.session(), entry.candidates().iter().map(|(_idx, candidate_hash)| *candidate_hash ).collect::<Vec<_>>()));
```

Comment:
This isn't per-se what it says in the guide. It would be accurate to call `candidates.clear()` every time `all_approved_max` gets set to `None`.

This would prevents us from passing too many blocks to the dispute coordinator later, when it really only needs to check the subchain between `lower_bound + 1 ..= all_approved_max`.